yt/yt/library/containers/cri/cri_executor.cpp

```diff
@@ -552,7 +571,11 @@ class TCriExecutor
         if (podSpec) {
             FillPodSandboxConfig(req->mutable_sandbox_config(), *podSpec);
         }
-        return req->Invoke().Apply(BIND([] (const TCriImageApi::TRspPullImagePtr& rsp) -> TCriImageDescriptor {
+
+        auto pullFuture = req->Invoke();
+        pullComplete.SetFrom(pullFuture);
+
+        return pullFuture.Apply(BIND([] (const TCriImageApi::TRspPullImagePtr& rsp) -> TCriImageDescriptor {
```

Comment:
In case if this future will be cancelled, all subscribers will be cancelled too. So, it's better to do
```
auto pullFuture = req->Invoke().ToUncancellable();
```

and after that ToUncancellable() above will not be required.

Reply:
This one must stay cancellable.

Reply:
Why is it better to retry pull in case of cancellation instead of ignoring it?

Reply:
Pulls are independent. Their callers are independent. They must break indepdently.
First pull could stuck and fail becuse of network issues.

Second and following pulls may waits optimiscaly to not generate redundant workload and result for free, but they needs this image too -- for more robust process next attermp they should start pulling from scratch.

I don't see reason to optimize weird case when first operation cancels pull but there are following operaions which could inherit it. Anyway, if the are no following operations pull must be canceled.

Reply:
Olay, that's reasonable!